### PR TITLE
Server: pass full url string and data to handleRequest()

### PR DESCRIPTION
**Pull request by dmarteau**
## Description
Pass the full url and data to the QgsServer::HandleRequest method. This eliminates the need
to pass request infos via environment variables which is no more required nor used when processing request. This also should fixe incomplete url's in the xref for getcapabilities, resulting in invalid layers.

### CODE REVIEW

File: tests/src/python/test_qgsserver_accesscontrol.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
+#t_wfstransaction_update/-*- coding: utf-8 -*-
```
Comment (sbrunner) [inline tests/src/python/test_qgsserver_accesscontrol.py]: ... ?

File: tests/src/python/test_qgsserver_accesscontrol.py [hunk 2]
```diff
@@ -1453,11 +1443,12 @@ def _img_diff_error(self, response, headers, image, max_diff=10, max_size_diff=Q
 
         self.assertTrue(test, message)
 
+
     def _geo_img_diff(self, image_1, image_2):
         if os.name == 'nt':
             # Not supported on Windows due to #13061
             return 0
-
+
```
Comment (sbrunner) [inline tests/src/python/test_qgsserver_accesscontrol.py]: Extra spaces...

**Review by sbrunner — APPROVED** [review]
Just some typo, also that's looks good :-)

**Comment by dmarteau** [comment]
@elpaso it's good for you ?

**Review by rldhont — APPROVED** [review]

**Comment by elpaso** [comment]
@dmarteau Thank! LGTM but please leave me some more time for review ... I have some minor questions but I don't have time right now.

**Comment by dmarteau** [comment]
May be there will some tests to add for checking the  xref validity f for getcapabilities.

**Comment by elpaso** [comment]
@dmarteau I think it's easier if we merge this PR as it is and I will propose some changes in another PR.

**Comment by elpaso** [comment]
@dmarteau @sbrunner I had to disable the check after this merge, can you please check what happened? https://travis-ci.org/qgis/QGIS/jobs/224604203#L1321

See: https://github.com/qgis/QGIS/commit/d35bd4abaa16baa5fc191d9435c788138d578bfc (hopefully passes)

**Comment by dmarteau** [comment]
@elpaso, I don't know, there was very little changes in this test. What I would like to know is why these tests passed the first time in the PR (before merging) ?

On OSX these tests are crashing (segfault): I compile with WITH_SPATIALITE and actually, any attempt to use spatialite bindings leads to segfault - something te be investigated.

**Comment by elpaso** [comment]
@dmarteau Travis is completely​ indeterministic :(

**Pull request merged**